File: KadirliApp/KadirliApp/Domain/Entities/DeathNotice.swift
import Foundation

struct DeathNotice: Identifiable, Codable {
    let id: UUID
    let firstName: String
    let lastName: String
    let deathDate: String // Supabase DATE tipi bazen String gelebilir, duruma göre Date yapılabilir.
    let burialPlace: String?
    let burialTime: String? // TIME tipi String olarak gelir (HH:mm:ss)
    let condolenceAddress: String?
    let latitude: Double?
    let longitude: Double?
    let imageUrl: String?
    let createdAt: Date
    
    var fullName: String {
        return "\(firstName) \(lastName)"
    }
    
    enum CodingKeys: String, CodingKey {
        case id
        case firstName = "first_name"
        case lastName = "last_name"
        case deathDate = "death_date"
        case burialPlace = "burial_place"
        case burialTime = "burial_time"
        case condolenceAddress = "condolence_address"
        case latitude, longitude
        case imageUrl = "image_url"
        case createdAt = "created_at"
    }
}

File: KadirliApp/API_Deneme/Models/DeathNotice.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace API_Deneme.Models
{
    public class DeathNotice
    {
        [Key]
        public Guid Id { get; set; }
        
        [Required]
        [Column("first_name")]
        [MaxLength(100)]
        public string FirstName { get; set; } = string.Empty;
        
        [Required]
        [Column("last_name")]
        [MaxLength(100)]
        public string LastName { get; set; } = string.Empty;
        
        [Required]
        [Column("death_date")]
        [MaxLength(10)]
        public string DeathDate { get; set; } = string.Empty; // YYYY-MM-DD formatında
        
        [Column("burial_place")]
        [MaxLength(200)]
        public string? BurialPlace { get; set; }
        
        [Column("burial_time")]
        [MaxLength(8)]
        public string? BurialTime { get; set; } // HH:mm:ss formatında
        
        [Column("condolence_address")]
        [MaxLength(500)]
        public string? CondolenceAddress { get; set; }
        
        public double? Latitude { get; set; }
        
        public double? Longitude { get; set; }
        
        [Column("image_url")]
        [MaxLength(500)]
        public string? ImageUrl { get; set; }
        
        [Column("created_at")]
        public DateTime CreatedAt { get; set; }
        
        [NotMapped]
        public string FullName => $"{FirstName} {LastName}";
    }
}

